apps/spruce/src/components/TaskBox/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { css } from "@emotion/react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import { Link } from "react-router-dom";
+import { TaskStatus } from "@evg-ui/lib/types/task";
+import { statusColorMap, statusIconMap } from "./icons";
+
+const { white } = palette;
+
+export const SQUARE_SIZE = 16;
+export const SQUARE_BORDER = 1;
+
+export const getTaskStatusStyle = (status: TaskStatus) => {
+  const icon = statusIconMap?.[status];
+  const iconStyle = icon ? `background-image: ${icon};` : "";
+  return css`
+    ${iconStyle}
+    background-color: ${statusColorMap[status]};
+  `;
+};
+
+export const TaskBox = styled(Link)<{ status: TaskStatus }>`
```

Comment:
So I noticed we have essentially 3 versions of this type of component.
This one.
[TaskStatsTooltip](https://github.com/evergreen-ci/ui/blob/6432255d25a5c676e491edf232335ba1c26711d7/apps/spruce/src/pages/waterfall/TaskStatsTooltip.tsx#L123)
[Waterfall Square](https://github.com/evergreen-ci/ui/blob/d25c9f2a4643e7edee2382f441c59f50dd37b376/apps/spruce/src/pages/waterfall/BuildRow.tsx#L251) Which seems to just be this one with some added tooltip logic. 

I think it would make sense to consolidate the 3 implementations so that we only have one instead of introducing an additional one. I also think the tooltip functionality for the waterfall would work really well here. 

https://github.com/user-attachments/assets/bbd5b2ee-6e45-47f7-8ab7-d441a9355069

Reply:
I'm not convinced of the usefulness of the tooltip since the same information will be present directly below; I think we can revisit this idea at another time. I added the hover mechanism that Will specified in his designs in place of this